operator/pkg/gateway-api/gateway_reconcile.go

```diff
@@ -370,3 +393,27 @@ func isReferenceAllowed(ctx context.Context, c client.Client, gw *gatewayv1beta1
 	}
 	return false, nil
 }
+
+func validateTLSSecret(ctx context.Context, c client.Client, namespace, name string) error {
+	secret := &corev1.Secret{}
+	if err := c.Get(ctx, client.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}, secret); err != nil {
+		return err
+	}
+
+	if !isValidPemFormat(secret.Data[corev1.TLSCertKey]) {
+		return fmt.Errorf("invalid certificate")
+	}
+
+	if !isValidPemFormat(secret.Data[corev1.TLSPrivateKeyKey]) {
+		return fmt.Errorf("invalid private key")
+	}
+	return nil
+}
+
+func isValidPemFormat(b []byte) bool {
```

Comment:
Nit: Technically, a PEM containing more than one block is _valid_, but I don't know if we want to accept it. (If there's more than one block, `rest` will have the other blocks as well).

Reply:
good point :100: